src/main/java/edu/harvard/iq/dataverse/api/DownloadInstanceWriter.java

```diff
@@ -585,4 +628,78 @@ private long getFileSize(DownloadInstance di, String extraHeader) {
         }
         return -1;
     }
+
+    /**
+     * @param range "bytes 0-10" for example. Found in the "Range" HTTP header.
+     * @param fileSize File size in bytes.
+     * @throws RunTimeException on any problems processing the Range header.
+     */
+    public List<Range> getRanges(String range, long fileSize) {
+        // Inspired by https://gist.github.com/davinkevin/b97e39d7ce89198774b4
+        // via https://stackoverflow.com/questions/28427339/how-to-implement-http-byte-range-requests-in-spring-mvc/28479001#28479001
+        List<Range> ranges = new ArrayList<>();
+
+        if (range != null) {
+            logger.fine("Range header supplied: " + range);
+
+            // Technically this regex supports multiple ranges.
+            // Below we have a check to enforce a single range.
+            if (!range.matches("^bytes=\\d*-\\d*(,\\d*-\\d*)*$")) {
+                throw new RuntimeException("The format is bytes=<range-start>-<range-end> where start and end are optional.");
+            }
+
+            if (ranges.isEmpty()) {
```

Comment:
I was confused by this line - at first I thought it was missing a "!" - but on a closer look it appears to be completely redundant :)

Reply:
Huh. I think you're right that it is redundant. I'll put this on the list.